api/src/Security/OAuth/HitobitoAuthenticator.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Security\OAuth;
+
+use App\Entity\User;
+use Doctrine\ORM\EntityManagerInterface;
+use KnpU\OAuth2ClientBundle\Client\ClientRegistry;
+use KnpU\OAuth2ClientBundle\Security\Authenticator\OAuth2Authenticator;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\PassportInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+
+class HitobitoAuthenticator extends OAuth2Authenticator {
+    public function __construct(
+        private AuthenticationSuccessHandlerInterface $authenticationSuccessHandler,
+        private ClientRegistry $clientRegistry,
+        private EntityManagerInterface $entityManager,
+        private RouterInterface $router,
+        private Security $security,
+    ) {
+    }
+
+    public function supports(Request $request): ?bool {
+        // continue ONLY if the current ROUTE matches the check ROUTE
+        return 'connect_hitobito_check' === $request->attributes->get('_route');
+    }
+
+    public function authenticate(Request $request): PassportInterface {
+        $client = $this->clientRegistry->getClient('hitobito');
+        $accessToken = $this->fetchAccessToken($client);
+
+        return new SelfValidatingPassport(
+            new UserBadge($accessToken->getToken(), function () use ($accessToken, $client) {
+                /** @var HitobitoUser $hitobitoUser */
+                $hitobitoUser = $client->fetchUserFromToken($accessToken);
+
+                $email = $hitobitoUser->getEmail();
+
+                $userRepository = $this->entityManager->getRepository(User::class);
+
+                // has the user logged in with Hitobito before?
+                $existingUser = $userRepository->findOneBy(['hitobitoId' => $hitobitoUser->getId()]);
```

Comment:
Die Datenbankspalte in der man das nachschaut müsste auch pro angebundene hitobito-Instanz unterschiedlich sein können. Könnte man die also auch noch entweder konfigurierbar machen oder das Finden in der Datenbank an den Hitobito-Provider (das ist ja der `$client` oder?) delegieren?